common/djangoapps/course_modes/tests/test_views.py

```diff
@@ -65,6 +65,25 @@ def test_redirect_to_dashboard(self, is_active, enrollment_mode, redirect):
         else:
             self.assertEquals(response.status_code, 200)
 
+    def test_no_id_redirect(self):
+        # Create the course modes
+        CourseModeFactory(mode_slug=CourseMode.NO_ID_PROFESSIONAL_MODES[0], course_id=self.course.id, min_price=100)
+
+        # Enroll the user in the test course
+        CourseEnrollmentFactory(
+            is_active=False,
+            mode=CourseMode.NO_ID_PROFESSIONAL_MODES[0],
```

Comment:
To echo Renzo's point above, I don't think it makes sense to use `NO_ID_PROFESSIONAL_MODES` as a list this way. If we're making assumptions about the ordering of the list, it doesn't seem useful to have it as a list.